### ISSUE TITLE
Failed to read the first frame of video file. That might mean that the file is corrupted

### ISSUE THREAD

**Issue by ring0jke** [Zulko/moviepy#1826]
```
from moviepy import *
import moviepy.video.fx.all as vfx
clip = VideoFileClip(path)
clip_final = clip.fx(vfx.time_mirror)

```

#### Expected Behavior
on line clip_final = clip.fx(vfx.time_mirror) it return exception MoviePy error: failed to read the first frame of video file /home/debian/PythonStuff/1672834326479.mp4. That might mean that the file is corrupted. That may also mean that you are using a deprecated version of FFMPEG. On Ubuntu/Debian for instance the version in the repos is deprecated. Please update to a recent version from the website.

So basically I haven't changed a code at all today. Todays it works perfectly. But right now it shows this error. 
Anyway to solve this bug or maybe some workaround?

#### Specifications

  - Python Version: Python 3.9.2
  - Moviepy Version: latest from pip
  - Platform Name: debian
  - Platform Version: debian latest

**Comment by ring0jke** [comment]
![win](https://user-images.githubusercontent.com/30176670/189482045-ea6d11ad-c299-4d4e-a7da-2ef1327c851a.png)


And now tried on my local machine on Windows. Got same resutlt. Any help?

**Comment by ring0jke** [comment]
The interesting part of it: 

File with size of 8 megabytes works while file with size of 18 mb throws error.

**Comment by Trahaswi** [comment]
The same bug for me too

**Comment by Trahaswi** [comment]
I need to try this in my arch computer

**Comment by Sammeeey** [comment]
I am running into the same error.
But not for all videos.
The issue for me is rather that I have no clue how to approach this issue.

Since it works for some videos, the FFMPEG version actually doesn't seem to be the problem.
Instead there might indeed be something wrong with the video file.
But the video works fine.
So I just don't know how to debug this!?

**Comment by James4Ever0** [comment]
This issue happens everywhere. Many of these claimed as a fix but as we can see it is not fixed at all. Take #1323 as an example.

**Comment by Implosiv3** [comment]
Hi. First of all, sorry about my English. I've been working with moviepy recently and ended up with this same error. I've been trying some code and I will share what I've found. I will investigate the moviepy source code later, but by now I say this to let you know.

I have a video that I programmatically create, and it last 3.5s exactly with 60fps (I know because I built the code that creates the video). I open it in any editor and it says 3.5s and 210 frames, so this is correct. If I do `VideoFileClip('video.mp4')` and print the `duration`, magically it says 3.61s (which is not correct). And this is the main reason of this issue.

If my video is 60fps and 3.5s, each frame lasts 0.0166...7s (1s / 60fps), so 3.5s - 0.0166...7s = 3.4833333s.

If I try to do `subclip(3.48, 3.61)` it works, if I try `subclip(3.49, 3.61)` it fails. I think (as I said, I will investigate the library source code later) that 3.49 is rounded up to 3.50, that is actually the last real clip frame, so the library throws an Exception when you try to subclip just the end `subclip(real_end_duration, real_end_duration)` with real and valid video duration. I mean, for the previous video you cannot do `subclip(3.50, 3.50)` because it throws and Exception, but you can do `subclip(3.50 - 1/60, 3.50)`, and that is the reason why the `subclip(3.49, 3.61)` fails, because it jumps to the next frame time that is 3.50, as 3.49 does not exist, and the `subclip(3.50, 3.61)` is not valid because 3.50 is the actual video `t_end` that you cannot use as `t_start`.

I've tried with another video that lasts 2.016666666s (121 frames, validated in video editor). The moviepy `.duration` shows a value of 2.02, which is rounded and wrong, as the real is 2.01666666s. So, even if I try to do `.subclip(duration - 1/60, duration)` it fails, because this is actually `.subclip(2.02 - 1/60, 2.02)`, and the `t_start`is not exactly 2.0, which would be accepted, it is 2.003333 that fails because it jumps to the next frame that is 2.016666666s, which is the last one and, as I said, not accepted as `t_start`. 

So, at the end, the `duration` parameter is rounded so be careful with it. A solution could be doing `duration= video.duration - (video.duration % (1 / 60))` to obtain the unrounded duration, but that doesn't solve the problem because the `duration` could be not the real one as it happens with the first video I mentioned.

**Comment by Implosiv3** [comment]
Hi. I'm back. I'm going to read the library code, but by now I've build my own solution:



    frame_time = 1 / video.fps
    for frame_number in reversed(range(int(video.fps * video.duration))):
        try:
            real_duration = frame_number * frame_time
            video.subclip(real_duration, 9999)
            # 't_start' can't be the last frame but the penultimate
            # so the real duration is one 'frame_time' later
            real_duration += frame_time
            break
        except:
            pass

    video = video.subclip(0, real_duration)

    return video

I don't like this code very much, but I can work with those 'corrupted' videos. I force subclipping until it doesn't throw any exception (this moment is when it is actually valid). Slow, ugly, but a temporary fix meanwhile I check the library code.

**Comment by digvijay7** [comment]
Faced the same issue. Somewhere over the internet(can't remember where) I found a solution for this:

```
@requires_duration
@apply_to_mask
@apply_to_audio
def custom_time_mirror(self):
  """
  Returns a clip that plays the current clip backwards.
  The clip must have its ``duration`` attribute set.
  The same effect is applied to the clip's audio and mask if any.
  """
  duration_per_frame = 1/self.fps
  return self.fl_time(lambda t: np.max(self.duration - t - duration_per_frame, 0), keep_duration=True)
```

This worked for me. Thanks

**Comment by OsaAjani** [comment]
Thank you for your contributions and for reporting issues in this repository. With the release of v2, which introduces significant changes to the codebase and API, we’ve reviewed the backlog of open PRs and issues. Due to the length of the backlog and the likelihood that many of these are either fixed or no longer applicable, we’ve made the decision to close all previous PRs and issues.

If you believe that any of these are still relevant to the current version or if you'd like to reopen a related discussion, please feel free to create a new issue or pull request, referencing the old one.

Thank you for your understanding and continued support!

**Comment by skrishnabharat** [comment]
I'm trying to extract audio from sas url . which is working with windows but somehow into not working in Linux. 
   
# Extract audio from video
    audio_path = f"{base_video_path}.mp3"
    clip = VideoFileClip(video_path)
    clip.audio.write_audiofile(audio_path, bitrate="32k")
    clip.audio.close()
    clip.close()

I'm getting this error, any suggestion is most welcome 

error:

OSError: MoviePy error: failed to read the first frame of video file. That might mean that the file is corrupted. That may also mean that you are using a deprecated version of FFMPEG. On Ubuntu/Debian for instance the version in the repos is deprecated. Please update to a recent version from the website.

**Comment by sayakpaul** [comment]
@digvijay7 do we know where the function should be applied?

**Comment by sayakpaul** [comment]
I can confirm that with the latest imageio-ffmpeg and moviepy versions, this didn't seem to be a problem for me.
